### ISSUE TITLE
Search Block: Consider adding a hidden button for "No button" option 

### ISSUE THREAD

**Issue by TeBenachi** [WordPress/gutenberg#38924]
### Description

The [accessibility guideline](https://make.wordpress.org/themes/handbook/review/accessibility/required/#forms) requires all forms to have the submit buttons. When selecting the “No Button” option in the Search block, the submit button is missing. 

### Step-by-step reproduction instructions

1. Select Search block
2. Select "No Button" option

### Screenshots, screen recording, code snippet

![no-button](https://user-images.githubusercontent.com/30734209/154774993-bb7789a1-98d8-45dc-8f13-8cce56597793.jpg)

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes
